### PR TITLE
Update Python Part of Iroha 

### PR DESCRIPTION
**Pull request by igor-egorov**
Signed-off-by: Igor Egorov <igor@soramitsu.co.jp>

### Description of the Change

Remove old Python lib.
Implement new Python lib (including crypto and network parts).
Update BTF examples.
Update tx-example.py.
Update permissions doc compiler.

### Benefits

Up to date Python lib.

### Possible Drawbacks 

Huge pr (due to outdated source branch).

### Usage Examples or Tests

```
binary_test
```

### CODE REVIEW

File: test/integration/binary/CMakeLists.txt [hunk 1]
```diff
@@ -3,42 +3,48 @@
...
-    compile_proto_to_python("${item}.proto")
-    list(APPEND PROTO_SWIG_DEPS "${SWIG_BUILD_DIR}/${item}_pb2.py")
-  endforeach (item)
+add_executable(binary_test
+    launchers.cpp
+    binaries_test.cpp
+    )
+target_link_libraries(binary_test
+    gtest::main
+    shared_model_proto_backend  # probably not needed here
```
Comment (igor-egorov) [inline test/integration/binary/CMakeLists.txt]: todo verify

**Comment by neewy** [comment]
Can we close then https://github.com/hyperledger/iroha/pull/1861 ?

File: example/python/batch-example.py [hunk 2]
```diff
@@ -4,16 +4,21 @@
...
 
 PLEASE ENSURE THAT MST IS ENABLED IN IROHA CONFIG
 
 """)
 
-from irohalib import Iroha, IrohaGrpc
-from irohalib import IrohaCrypto as ic
 
-import binascii
+if sys.version_info[0] < 3:
```
[the remark this answers is not in the document]
Comment (igor-egorov) [reply example/python/batch-example.py]: Should be decided on a community meeting.

File: example/python/irohalib.py [hunk 3]
```diff
@@ -44,7 +44,13 @@ def hash(proto_with_payload):
         :proto_with_payload: proto transaction or query
         :return: bytes representation of hash
         """
-        bytes = proto_with_payload.payload.SerializeToString()
+        obj = None
+        if hasattr(proto_with_payload, 'payload'):
+            obj = getattr(proto_with_payload, 'payload')
+        elif hasattr(proto_with_payload, 'meta'):
```
[the remark this answers is not in the document]
Comment (igor-egorov) [reply example/python/irohalib.py]: Added a comment. That is for special kind of queries (block streaming queries) which do not have a payload field.

File: example/python/ed25519.py [hunk 4]
```diff
@@ -43,6 +43,9 @@
 import operator
 import sys
 
+if sys.version_info < (3, 6):
+    import sha3
+
```
Comment (muratovv) [inline example/python/ed25519.py]: Where is `else` branch?

File: example/python/ed25519.py [hunk 5]
```diff
@@ -43,6 +43,9 @@
 import operator
 import sys
 
+if sys.version_info < (3, 6):
```
Comment (l4l) [inline example/python/ed25519.py]: Ehm, what about `>= (3, 6)`, ain't it require an import as well?

Comment (igor-egorov) [reply example/python/ed25519.py]: You are right. Since 3.6 no additional imports are required.

File: CMakeLists.txt [hunk 6]
```diff
@@ -71,16 +71,15 @@ SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
 option(BENCHMARKING "Build benchmarks"               OFF)
 option(TESTING      "Build tests"                    ON )
+option(USE_BTF      "Build Binary Testing Framwork"  OFF)
```
Comment (paolodina) [inline CMakeLists.txt]: typo

File: docs/source/guides/build.rst [hunk 7]
```diff
@@ -25,13 +25,13 @@ Docker
 .. note:: You don't need Docker to run Iroha, it is just one of the possible
   choices.
 
-First of all, you need to install ``docker`` and ``docker-compose``. You can 
-read how to install it on a 
+First of all, you need to install ``docker`` and ``docker-compose``. You can
+read how to install it on a
```
Comment (paolodina) [inline docs/source/guides/build.rst]: s/on a/on the ?

**Pull request merged**